File: runtime/JavaScript/src/antlr4/error/InputMismatchException.js
/* Copyright (c) 2012-2022 The ANTLR Project Contributors. All rights reserved.
 * Use is of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */
import RecognitionException from "./RecognitionException.js";

/**
 * This signifies any kind of mismatched input exceptions such as
 * when the current input does not match the expected token.
 */
export default class InputMismatchException extends RecognitionException {
    constructor(recognizer) {
        super({message: "", recognizer: recognizer, input: recognizer.getInputStream(), ctx: recognizer._ctx});
        this.offendingToken = recognizer.getCurrentToken();
    }
}

File: runtime/CSharp/src/InputMismatchException.cs
/* Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */

namespace Antlr4.Runtime
{
    /// <summary>
    /// This signifies any kind of mismatched input exceptions such as
    /// when the current input does not match the expected token.
    /// </summary>
    /// <remarks>
    /// This signifies any kind of mismatched input exceptions such as
    /// when the current input does not match the expected token.
    /// </remarks>
    [System.Serializable]
    public class InputMismatchException : RecognitionException
    {
        public InputMismatchException(Parser recognizer)
			: base(recognizer, ((ITokenStream)recognizer.InputStream), recognizer.RuleContext)
        {
            OffendingToken = recognizer.CurrentToken;
        }
    }
}
